src/librustc_parse/parser/diagnostics.rs

```diff
@@ -934,6 +934,25 @@ impl<'a> Parser<'a> {
             return self.expect(&token::Semi).map(drop);
         } else if !sm.is_multiline(self.prev_token.span.until(self.token.span)) {
             // The current token is in the same line as the prior token, not recoverable.
+        } else if sm.is_multiline(self.prev_token.span.until(self.token.span))
```

Comment:
The `is_multiline` is already implied by the previous if branch.

Reply:
Changed this locally. Will wait to hear your thoughts about the current set of changes and where we are with the report before I push anything. See https://github.com/rust-lang/rust/pull/72348#discussion_r427719055 for something that is very close to what we want.